### PR TITLE
added sustainability classifications to rules

### PR DESCRIPTION
**Pull request by stokpop**
First classification round, please check and verify.

This is for issue #306 

Find "performance" tags without sustainability tags: should it still be added?

Blocked threads issues (e.g. without timeout): cpu or memory?
Timeout issues: cpu or io?

Can sustainability tags also have multiple area's? e.g. cpu and memory, or even all three: cpu, memory, io?

### CODE REVIEW

File: src/main/resources/category/java/remoting.xml [hunk 1]
```diff
@@ -327,7 +328,8 @@ ancestor::MethodDeclaration//PrimarySuffix/@Image="disableConnectionState")]
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
             <property name="tag" value="suspicious" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-low" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/remoting.xml:331]: not sure about this, how does it impact sustainability?

File: src/main/resources/category/java/remoting.xml [hunk 2]
```diff
@@ -660,7 +662,8 @@ class Foo {
             <property name="tag" value="performance" type="String" description="for-sonar"/>
             <property name="tag" value="resilience" type="String" description="for-sonar"/>
             <property name="tag" value="suspicious" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-low" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/remoting.xml:665]: dit is informational, sustainability impact is niet duidelijk, zou ik weghalen.

File: src/main/resources/category/java/remoting.xml [hunk 3]
```diff
@@ -876,8 +877,9 @@ and use it to replace the bad line in Bad example:
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
+            <property name="tag" value="sustainability-medium" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/remoting.xml:880]: not sure about medium

File: src/main/resources/category/java/remoting.xml [hunk 4]
```diff
@@ -920,7 +922,8 @@ and use it to replace the bad line in Bad example:
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-medium" type="String" description="for-sonar"/>
+            <property name="tag" value="cpu" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/remoting.xml:926]: not sure about medium
also memory

File: src/main/resources/category/java/spring.xml [hunk 5]
```diff
@@ -197,7 +198,8 @@ and  (../../ClassOrInterfaceBodyDeclaration/Annotation//Name[@Image='Autowired']
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-low" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/spring.xml:201]: maybe medium

File: src/main/resources/category/java/spring.xml [hunk 6]
```diff
@@ -421,7 +424,6 @@ class Foo {
             <property name="tag" value="performance" type="String" description="for-sonar"/>
```
Comment (jborgers) [inline src/main/resources/category/java/spring.xml:424]: maybe not performance

File: src/main/resources/category/java/spring.xml [hunk 7]
```diff
@@ -480,7 +482,6 @@ class GoodCacheKeyGenerator implements KeyGenerator {
...
```
Comment (jborgers) [inline src/main/resources/category/java/spring.xml:482]: maybe not performance

**Review by jborgers — CHANGES_REQUESTED** [review]
heel wat werk

File: src/main/resources/category/java/common.xml [hunk 8]
```diff
@@ -1642,7 +1649,8 @@ Set<YourEnumType> set = EnumSet.allOf(YourEnumType.class);
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/common.xml:1652]: put on medium... toString() can be heavy when called *many* times for nothing

File: src/main/resources/category/java/common.xml [hunk 9]
```diff
@@ -1988,7 +1997,8 @@ class Foo {
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-high" type="String" description="for-sonar"/>
+            <property name="tag" value="cpu" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/common.xml:2001]: yes

File: src/main/resources/category/java/common.xml [hunk 10]
```diff
@@ -2075,7 +2086,8 @@ class Foo {
         <properties>
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+            <property name="tag" value="sustainability-medium" type="String" description="for-sonar"/>
+            <property name="tag" value="memory" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/common.xml:2090]: yes

File: src/main/resources/category/java/concurrent.xml [hunk 11]
```diff
@@ -822,7 +808,6 @@ public class Foo {
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="multi-threading" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/concurrent.xml:810]: See other remark about timeouts for threads... it is a potential stability (and performance) issue...

File: src/main/resources/category/java/concurrent.xml [hunk 12]
```diff
@@ -898,8 +883,7 @@ and not (../PrimarySuffix[ends-with(@Image, 'Timeout')])]
...
-	    <property name="tag" value="unpredictable" type="String" description="for-sonar"/>
-            <property name="version" value="2.0"/>
+	        <property name="tag" value="unpredictable" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/concurrent.xml:886]: yes, or "stability"

File: src/main/resources/category/java/concurrent.xml [hunk 13]
```diff
@@ -1304,7 +1286,6 @@ public class FooBad {
             <property name="tag" value="jpinpoint-rule" type="String" description="for-sonar"/>
             <property name="tag" value="multi-threading" type="String" description="for-sonar"/>
             <property name="tag" value="performance" type="String" description="for-sonar"/>
```
[the remark this answers is not in the document]
Comment (stokpop) [reply src/main/resources/category/java/concurrent.xml:1288]: "stability"?

File: src/main/resources/category/java/remoting.xml (hunk 4 above)
Comment (stokpop) [reply src/main/resources/category/java/remoting.xml:926]: Memory added. Moved to high.

File: src/main/resources/category/java/spring.xml (hunk 7 above)
Comment (stokpop) [reply src/main/resources/category/java/spring.xml:481]: removed performance tag

**Comment by stokpop** [comment]
@jborgers can we go through to non-closed conversations together?

**Comment by jborgers** [comment]
Sure, Friday?

Op ma 17 jun 2024 om 09:54 schreef Peter Paul Bakker <
***@***.***>:

**Pull request merged**